src/Common/src/System/Net/Logging/SocketsLogging.cs

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Threading;
+
+namespace System.Net
+{
+    internal static class SocketsLogging
```

Comment:
We can actually share the common code. The [Working Effectively with Legacy Code](http://www.amazon.com/Working-Effectively-Legacy-Michael-Feathers/dp/0131177052) book describes the technique as code seaming.
Make Logging partial and refactor the Logging.cs code such that it's written as a template class. The actual definition of the eventSource will need to remain particular to each component (e.g. Logging.Sockets.cs).